### PR TITLE
checkmake pre-commit hook

### PR DESCRIPTION
**Pull request by joey0320**
Add checkmake as pre-commit. 

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [x] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [x] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,3 +9,7 @@ repos:
       - id: check-yaml
       - id: check-json
       - id: check-added-large-files
+  - repo: https://github.com/ucb-bar/checkmake.git
```
Comment (jerryz123) [inline .pre-commit-config.yaml:12]: Why use a fork instead of upstream?

Comment (joey0320) [reply .pre-commit-config.yaml:12]: The upstream checks if the makefile contains a bunch of default PHONYs & we don't need that.

Comment (jerryz123) [reply .pre-commit-config.yaml:12]: I suspect there's a way to pass args to the pre-commit hook defined in that repo. https://pre-commit.com/#new-hooks

Is there a way to pass a arg that disables the min-phony stuff?

Comment (joey0320) [reply .pre-commit-config.yaml]: Done!

**Review by jerryz123 — APPROVED** [review]
Thanks for looking into this!

**Comment by joey0320** [comment]
thxs for the review!

**Pull request merged**